File: cbc-py/test_cbc.py
import cbc_py
import sys

def main():
    artifact_path = "../test.cbc"
    
    print(f"Testing cbc-py with {artifact_path}...")

    # Test inspection
    try:
        info = cbc_py.inspect(artifact_path)
        print("\n[Inspection Result]")
        print(f"Valid Bootstrap: {info.valid_bootstrap}")
        print(f"Version: {info.version}")
        print(f"Hash Suite: {info.hash_suite}")
        print(f"Blocks: {info.block_count}")
        print(f"Block Size: {info.block_payload_size}")
        print(f"Families: {', '.join(info.families)}")
    except ValueError as e:
        print(f"Inspection failed: {e}")
        sys.exit(1)

    # Test validation
    try:
        is_valid = cbc_py.validate(artifact_path)
        print(f"\nValidation Status: {'✓ VALID' if is_valid else '✗ INVALID'}")
    except ValueError as e:
        print(f"Validation failed: {e}")
        sys.exit(1)

if __name__ == "__main__":
    main()

File: cbc-node/test_cbc.js
const { inspectFile, validateFile } = require('./index.node');
const path = require('path');

const artifactPath = path.resolve('../test.cbc');

console.log(`Testing cbc-node with ${artifactPath}...`);

try {
    const info = inspectFile(artifactPath);
    console.log('\n[Inspection Result]');
    console.log(`Valid Bootstrap: ${info.validBootstrap}`);
    console.log(`Hash Suite: ${info.hashSuite}`);
    console.log(`Block Size: ${info.blockPayloadSize}`);
    console.log(`Families: ${info.families.join(', ')}`);
} catch (err) {
    console.error(`Inspection failed: ${err.message}`);
    process.exit(1);
}

try {
    const isValid = validateFile(artifactPath);
    console.log(`\nValidation Status: ${isValid ? '✓ VALID' : '✗ INVALID'}`);
} catch (err) {
    console.error(`Validation failed: ${err.message}`);
    process.exit(1);
}
